components/dashboard/src/settings/Notifications.tsx

```diff
@@ -14,50 +14,65 @@ import settingsMenu from "./settings-menu";
 export default function Notifications() {
     const { user, setUser } = useContext(UserContext);
 
-    const [isTransactionalMail, setTransactionMail] = useState(!user?.additionalData?.emailNotificationSettings?.disallowTransactionalEmails);
-    const toggleTransactionalMail = async () => {
-        if (user) {
-            user.additionalData = {
-                ...{
+    const [isChangelogMail, setChangelogMail] = useState(!!user?.additionalData?.emailNotificationSettings?.allowsChangelogMail);
+    const toggleChangelogMail = async () => {
+        if (user && user.additionalData && user.additionalData.emailNotificationSettings) {
+            await getGitpodService().server.updateLoggedInUser({
+                additionalData: {
                     ...user.additionalData,
                     emailNotificationSettings: {
-                        ...user.additionalData?.emailNotificationSettings,
-                        disallowTransactionalEmails: isTransactionalMail
+                        ...user.additionalData.emailNotificationSettings,
+                        allowsChangelogMail: !isChangelogMail
                     }
                 }
-            }
-            await getGitpodService().server.updateLoggedInUser({
-                additionalData: user.additionalData
             });
+            await getGitpodService().server.trackEvent({
+                event: "notification_change",
+                properties: { "unsubscribed_changelog": isChangelogMail }
```

Comment:
Nit: I was at first glance confused why you're not using the `!isChangelogMail` here. Maybe it'd be better readable if we create a variable `newIsChangelogMail = !isChangelogMail` at the beginning of this block and use that instead of `!isChangeLogMail` in the following and write `!newIsChangelogMail` here. WDYT?

Reply:
Yes this part did get confusing to me too, because the event name is a negative word “unsubscribe”. But your suggestion makes it a bit more readable. Will do this change.